Userland/Libraries/LibWeb/CSS/CSSStyleDeclaration.h

```diff
@@ -26,9 +26,11 @@ class CSSStyleDeclaration : public Bindings::PlatformObject {
     virtual String item(size_t index) const = 0;
 
     virtual Optional<StyleProperty> property(PropertyID) const = 0;
+    virtual Optional<StyleProperty> custom_property(FlyString const& custom_property_name) const = 0;
 
-    virtual WebIDL::ExceptionOr<void> set_property(PropertyID, StringView css_text, StringView priority = ""sv) = 0;
-    virtual WebIDL::ExceptionOr<String> remove_property(PropertyID) = 0;
+    // property_name only needs to be passed for custom properties.
+    virtual WebIDL::ExceptionOr<void> set_property(PropertyID, StringView property_name, StringView css_text, StringView priority = ""sv) = 0;
+    virtual WebIDL::ExceptionOr<String> remove_property(PropertyID, StringView property_name) = 0;
 
```

Comment:
have you tried to declare default value for `property_name` just like `priority`? although have to make it the second last or last argument to work:

```cpp
set_property(PropertyID, StringView css_text, StringView priority = ""sv, StringView property_name = ""sv)
```

Reply:
This refactor does make sense. I will try this out and put this up.